gammapy/datasets/map.py

```diff
@@ -2694,6 +2729,36 @@ def to_map_dataset(self, name=None):
             meta_table=self.meta_table,
         )
 
+    def _to_asimov_dataset(self, name=None, models=None):
+        """Create Asimov dataset"""
+        npred = self.npred()
+        data = np.nan_to_num(npred.data, copy=True, nan=0.0, posinf=0.0, neginf=0.0)
+        npred.data = data.astype("float")
```

Comment:
you can use `super` to avoid doing this twice?

Reply:
I don't think it is applicable here.